ClassCastException on moving to mysql-connector-java version 8.0.23

<!--- This environment context section helps us quickly review your PR. 
      Please take a minute to fill-out this information. -->

## Environment
**Liquibase Version**:
3.6.3

**Liquibase Integration & Version**: 
spring boot

**Liquibase Extension(s) & Version**: 

**Database Vendor & Version**:
mysql  Ver 14.14 Distrib 5.7.28, for osx10.14 (x86_64)
mysql-connector-java version 8.0.23

**Operating System Type & Version**:
All 

## Description
While trying to move from mysql-connector-java version 8.0.18 to 8.0.23, getting an ClassCastException on bootup.
`Caused by: java.lang.ClassCastException: java.time.LocalDateTime cannot be cast to java.lang.String
	at liquibase.changelog.StandardChangeLogHistoryService.getRanChangeSets(StandardChangeLogHistoryService.java:324)
	at liquibase.changelog.AbstractChangeLogHistoryService.upgradeChecksums(AbstractChangeLogHistoryService.java:66)
	at liquibase.changelog.StandardChangeLogHistoryService.upgradeChecksums(StandardChangeLogHistoryService.java:293)
	at liquibase.Liquibase.checkLiquibaseTables(Liquibase.java:1221)
	at liquibase.Liquibase.update(Liquibase.java:193)
	at liquibase.Liquibase.update(Liquibase.java:179)
	at liquibase.integration.spring.SpringLiquibase.performUpdate(SpringLiquibase.java:353)
	at liquibase.integration.spring.SpringLiquibase.afterPropertiesSet(SpringLiquibase.java:305)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.invokeInitMethods(AbstractAutowireCapableBeanFactory.java:1830)
	at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.initializeBean(AbstractAutowireCapableBeanFactory.java:1767)
	... 108 common frames omitted
`

## Steps To Reproduce (CLI)
Add the mysql-connector 8.0.23 jar to your lib directory and attempt to run liquibase operations against a mysql database.

## Steps To Reproduce (Spring Boot)
List the steps to reproduce the behavior.
this is the pom.xml used,

```

<dependency>
	<groupId>mysql</groupId>
	<artifactId>mysql-connector-java</artifactId>
	<version>8.0.23</version>
</dependency>

<dependency>
	<groupId>org.liquibase</groupId>
	<artifactId>liquibase-core</artifactId>
	<version>3.6.3</version>
</dependency>
<dependency>
	<groupId>org.liquibase.ext</groupId>
	<artifactId>liquibase-hibernate5</artifactId>
	<version>3.6</version>
	<exclusions>
		<exclusion>
			<groupId>org.hibernate.javax.persistence</groupId>
			<artifactId>hibernate-jpa-2.0-api</artifactId>
		</exclusion>
	</exclusions>
</dependency>

```

## Liquibase Internal QA Requirements
* Manually validate the fix with the CLi. 
* Locally execute all functional MySQL tests.



┆Issue is synchronized with this [Jira Bug](https://datical.atlassian.net/browse/LB-1131) by [Unito](https://www.unito.io/learn-more)
┆Fix Versions: Community 4.3.1,Liquibase 4.3.1

--- a/liquibase-core/src/main/java/liquibase/changelog/StandardChangeLogHistoryService.java
+++ b/liquibase-core/src/main/java/liquibase/changelog/StandardChangeLogHistoryService.java
@@ -30,6 +30,8 @@ import liquibase.structure.core.Table;
 import java.text.DateFormat;
 import java.text.ParseException;
 import java.text.SimpleDateFormat;
+import java.time.LocalDateTime;
+import java.time.ZoneId;
 import java.util.*;
 
 public class StandardChangeLogHistoryService extends AbstractChangeLogHistoryService {
@@ -318,6 +320,8 @@ public class StandardChangeLogHistoryService extends AbstractChangeLogHistorySer
                     Date dateExecuted = null;
                     if (tmpDateExecuted instanceof Date) {
                         dateExecuted = (Date) tmpDateExecuted;
+                    } else if (tmpDateExecuted instanceof LocalDateTime) {
+                        dateExecuted = Date.from(((LocalDateTime) tmpDateExecuted).atZone(ZoneId.systemDefault()).toInstant());
                     } else {
                         DateFormat df = new SimpleDateFormat("yyyy-MM-dd HH:mm:ss");
                         try {

